contracts/JoepegExchange.sol

```diff
@@ -560,6 +594,53 @@ contract JoepegExchange is
         }
     }
 
+    /**
+     * @notice Match multiple asks with their respective taker bid order using AVAX and WAVAX, ignoring expired maker asks
+     * @dev Used when the caller doesn't want the transaction to revert if a maker ask already expired
+     * @param trades an array of trades
+     */
+    function batchBuyWithAVAXAndWAVAXIgnoringExpiredAsks(
+        Trade[] calldata trades
+    ) external payable nonReentrant {
+        // Calculate the total cost of all valid orders
+        uint256 totalCost;
+        for (uint256 i; i < trades.length; ++i) {
+            if (_validateMakerOrder(trades[i].makerAsk)) {
+                totalCost += trades[i].takerBid.price;
+            }
+        }
+
+        // Transfer WAVAX if needed
+        if (totalCost > msg.value) {
+            IERC20(WAVAX).safeTransferFrom(
+                msg.sender,
+                address(this),
+                (totalCost - msg.value)
+            );
+
+            // Wrap AVAX sent to this contract
+            IWAVAX(WAVAX).deposit{value: msg.value}();
+        } else {
+            // Wrap AVAX needed to pay for valid orders
+            IWAVAX(WAVAX).deposit{value: totalCost}();
+        }
+
+        // Match orders
+        for (uint256 i; i < trades.length; ++i) {
+            _matchAskWithTakerBidUsingAVAXAndWAVAXIgnoringExpiredAsks(
+                trades[i].takerBid,
+                trades[i].makerAsk
+            );
+        }
```

Comment:
instead of iterating through all the trades here again and doing validation again on the ones that are expired, could we construct an array of valid trades when we first iterate through them on lines 607-610 and just iterate through those here? then I don't think you'd need `_matchAskWithTakerBidUsingAVAXAndWAVAXIgnoringExpiredAsks`

Reply:
Yeah I tried this at first but I'm not sure how to create a dynamic array in the scope of the function and push valid trades to it ?

Reply:
Yeah it's not easily doable

Reply:
would it be better to perform the expired order check directly here so you would not need an additional `_matchAskWithTakerBidUsingAVAXAndWAVAXIgnoringExpiredAsks` method? i notice that `_validateMakerAskAndTakerBid` is done twice when calling this method too - once in the method, and again in the call to the `_matchAskWithTakerBidUsingAVAXAndWAVAX` method. 

so perhaps something like
```
for (uint256 i; ...) {
    if (_checkMakerOrderNotExpired(trades[i].makerAsk)) {
        _matchAskWithTakerBidUsingAVAXAndWAVAX(...);
    }
}
```

you could also keep track of the boolean return values here if needed.